test/unit/core/connection_string.test.js

```diff
@@ -284,6 +286,48 @@ describe('Connection String', function () {
               });
             }
           });
+
+          it(`${test.description} -- new MongoOptions parser`, function () {
```

Comment:
There's another ticket to integrate it into the code base, right now parseOptions is called in the MongoClient constructor, I was thinking parseConnectionString, (naming aside) will just conditionally do the async stuff remaining namely the srv lookup and be passed the parsed options.
The test needs a little reworking anyway because MongoOptions has a different shape.